composer/loggers/mosaicml_logger.py

```diff
@@ -96,10 +125,104 @@ def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
     def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
         self._log_metadata(metrics)
 
+    def log_analytics(self,) -> None:
+        if self.analytics_data is None:
+            return
+
+        # Fetch / cast metrics that we want to log from self.analytics_data
+        autoresume: bool = self.analytics_data['autoresume']
+        trainer_state: State = self.analytics_data['state']
+        save_interval: Union[str, int, Time, Callable[[State, Event], bool]] = self.analytics_data['save_interval']
+        loggers: List[LoggerDestination] = self.analytics_data['loggers']
+        load_path: Union[str, None] = self.analytics_data['load_path']
+        save_folder: Union[str, None] = self.analytics_data['save_folder']
+
+        metrics: Dict[str, Any] = {'composer/autoresume': autoresume, 'composer/precision': trainer_state.precision}
+
+        train_dataloader = trainer_state.train_dataloader
+        if train_dataloader is not None and isinstance(train_dataloader, torch.utils.data.DataLoader):
+            metrics['composer/train_loader_workers'] = train_dataloader.num_workers
+
+        metrics['composer/eval_loaders'] = []
+        for evaluator in trainer_state.evaluators:
+            dataloader = evaluator.dataloader.dataloader
+            if isinstance(dataloader, torch.utils.data.DataLoader):
+                metrics['composer/eval_loaders'].append(
+                    json.dumps({
+                        'label': evaluator.label,
+                        'num_workers': dataloader.num_workers,
+                    }),
+                )
+
+        get_optimizer_args = lambda optimizer: {
+            k: v for k, v in optimizer.__dict__.items() if not k.startswith('_') and k != 'param_groups'
+        }
+        metrics['composer/optimizers'] = [{
+            optimizer.__class__.__name__: get_optimizer_args(optimizer),
+        } for optimizer in trainer_state.optimizers]
+        metrics['composer/algorithms'] = [{
+            algorithm.__class__.__name__: algorithm.__dict__,
```

Comment:
I think you probably don't want to log `.__dict__`, as there is a lot of stuff in the `.__dict__` of a python object.

Reply:
will just use `optimizer.defaults` instead of `__dict__`, but what do you think is a good alternative for `__dict__` when it comes to subclasses of `Algorithm`? do you think `__repr__` would be a good alternative?

Reply:
I think its ok to just get the class name